step-functions-data-science-sdk/automate_model_retraining_workflow/automate_model_retraining_workflow.ipynb

```diff
@@ -0,0 +1,983 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "project_name = 'ml_deploy'\n",
+    "\n",
+    "data_source = sagemaker.s3.S3Uploader.upload(local_path='./data/customer-churn.csv',\n",
+    "                               desired_s3_uri='s3://{}/{}'.format(bucket, project_name),\n",
+    "                               session=session)\n",
+    "\n",
+    "train_prefix = 'train'\n",
+    "val_prefix = 'validation'\n",
+    "\n",
+    "train_data = 's3://{}/{}/{}/'.format(bucket, project_name, train_prefix)\n",
+    "validation_data = 's3://{}/{}/{}/'.format(bucket, project_name, val_prefix)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Resources\n",
+    "In the following steps we'll create the Glue job and Lambda function that are called from the Step Functions workflow."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create the AWS Glue Job"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "glue_script_location = sagemaker.s3.S3Uploader.upload(local_path='./code/glue_etl.py',\n",
+    "                               desired_s3_uri='s3://{}/{}'.format(bucket, project_name),\n",
+    "                               session=session)\n",
+    "glue_client = boto3.client('glue')\n",
+    "\n",
+    "response = glue_client.create_job(\n",
+    "    Name=job_name,\n",
+    "    Description='PySpark job to extract the data and split in to training and validation data sets',\n",
+    "    Role=glue_role, # you can pass your existing AWS Glue role here if you have used Glue before\n",
+    "    ExecutionProperty={\n",
+    "        'MaxConcurrentRuns': 2\n",
+    "    },\n",
+    "    Command={\n",
+    "        'Name': 'glueetl',\n",
+    "        'ScriptLocation': glue_script_location,\n",
+    "        'PythonVersion': '3'\n",
+    "    },\n",
+    "    DefaultArguments={\n",
+    "        '--job-language': 'python'\n",
+    "    },\n",
+    "    GlueVersion='1.0',\n",
+    "    WorkerType='Standard',\n",
+    "    NumberOfWorkers=2,\n",
+    "    Timeout=60\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create the AWS Lambda Function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import zipfile\n",
+    "zip_name = 'query_training_status.zip'\n",
+    "lambda_source_code = './code/query_training_status.py'\n",
+    "\n",
+    "zf = zipfile.ZipFile(zip_name, mode='w')\n",
+    "zf.write(lambda_source_code, arcname=lambda_source_code.split('/')[-1])\n",
+    "zf.close()\n",
+    "\n",
+    "\n",
+    "sagemaker.s3.S3Uploader.upload(local_path=zip_name,\n",
```

Comment:
[Minor] Consider importing this so it's cleaner to write:
```
from sagemaker.s3 import S3Uploader
data_source = S3Uploader.upload([...])
```

Reply:
Good idea. Resolved in the most recent commit.